sympy/algebras/quaternion.py

```diff
@@ -39,7 +39,7 @@ def __new__(cls, a=0, b=0, c=0, d=0):
         c = sympify(c)
         d = sympify(d)
 
-        if any(i.is_real is False for i in [a, b, c, d]):
+        if any(i.is_commutative is False for i in [a, b, c, d]):
             raise ValueError("arguments have to be real")
```

Comment:
real -> commutative

Reply:
I didn't get you. U suggested just checking for commutative right?

Reply:
In the error message as well.

Reply:
Sorry about that. I changed it.